src/materials/H5MoltresMaterial.C

```diff
@@ -0,0 +1,480 @@
+#include "H5MoltresMaterial.h"
```

Comment:
There is a massive amount of repeated code between H5MoltresMaterial and MoltresMaterial. I suggest that you make an [abstract base class](https://www.ibm.com/support/knowledgecenter/SS2LWA_12.1.0/com.ibm.xlcpp121.bg.doc/language_ref/cplr142.html) which has the parameter specifications and selection of XS interpolation method and associated methods, and inherit from that for the normal MoltresMaterial and H5MoltresMaterial. If that doesn't make sense, I can give you a more clear idea of what it should look like. This will cut the number of lines of code for this about in half, and make it more easily maintainable.

Reply:
I agree.